packages/core/src/mixer.ts

```diff
@@ -1,64 +1,111 @@
-import HeapPackage, { type default as HeapType } from 'heap-js'
+import HeapPackage from 'heap-js'
+// Issues with ESM;
+// Code and types are loaded differently
+import type { Heap as HeapType } from 'heap-js'
+
 import { randomInteger } from '@hoprnet/hopr-utils'
 import type { Packet } from './messages/index.js'
 import { MAX_PACKET_DELAY } from './constants.js'
+
+// @ts-ignore untyped package
+import retimer from 'retimer'
+
 //import debug from 'debug'
 //const log = debug('hopr-core:mixer')
 
-type HeapElement = [number, Packet]
+type MixerEntry = [number, Packet]
 
-const comparator = (a: HeapElement, b: HeapElement): number => {
-  if (b[0] < a[0]) {
-    return 1
-  } else if (b[0] > a[0]) {
-    return -1
-  }
-  return 0
+const comparator = (a: MixerEntry, b: MixerEntry): number => {
+  return a[0] - b[0]
```

Comment:
This should rather use `signum` to stay within [-1,1] interval.

Reply:
Don't think it's necessary according to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#description